core/lib/instrument.c

```diff
@@ -6799,7 +6799,7 @@ dr_get_mcontext_priv(dcontext_t *dcontext, dr_mcontext_t *dmc, priv_mcontext_t *
     else if (TEST(DR_MC_CONTROL, dmc->flags))
         dmc->xsp = get_mcontext(dcontext)->xsp;
 
-#ifdef ARM
+#ifdef AARCHXX
     if (TEST(DR_MC_INTEGER, dmc->flags)) {
         /* get the stolen register's app value */
         if (mc != NULL) {
```

Comment:
It would also be nice to have a test that fails without each of the get and set fixes here (not to pile too much work on this...)

Reply:
I should be able to create a test that mimic the original bug I found. Hopefully that will catch the absence of the other fix too, but will have to look into it.